components/FavoriteButton.tsx

```diff
@@ -0,0 +1,46 @@
+import FilledHeart from '@/public/icons/ic_fluent_heart_24_filled.svg';
+import styled, { css } from 'styled-components';
+
+export default function FavoriteButton({
+  $isDetailPage,
+  onToggleFavorite,
+  isFavorite,
+}) {
+  return (
+    <>
+      <StyledButton
+        $isDetailPage={$isDetailPage}
+        onClick={onToggleFavorite}
+        $isFavorite={isFavorite}
+      >
+        <FilledHeart />
+      </StyledButton>
+    </>
+  );
+}
+
+const StyledButton = styled.button`
```

Comment:
You aren't typing your styled components. This is the approach from the styled-components docs:

```js

interface LinkProps {
  readonly $isDetailPage?: boolean;
  readonly $isFavorite?: boolean;
}

const StyledButton = styled.button<LinkProps>`
your CSS comes here
` 

```